ls1intum/Artemis
# `Programming exercises`: Display hints in online code editor

### Checklist
#### General
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
Introduced in #5136, the hint system in the client has been reworked to display hints dynamically. These hints were only accessible from the student's exercise participation screen, but not from the code editor.

### Description
These changes add the same UI elements that are visible from the 's exercise participation page to the code editor.

### Steps for Testing
Prerequisites:
- 1 Programming Exercise with Hints and Online Code Editor

1. Log in to Artemis
2. Navigate to the programming exercise
3. Open the online code editor
4. Follow the testing steps to display a hint dynamically as described in #5136 by making at least three unsuccessful submissions.
5. Verify that a button is displayed that can be accessed for new hints, or if a hint has already been activated.
6. Verify that an alert is shown for new hints.
7. Make sure that no hints are displayed in the exam mode.

### Review Progress
#### Code Review
#### Manual Tests

### Screenshots
![Bildschirmfoto 2022-06-05 um 16 33 15](https://user-images.githubusercontent.com/73833780/172055701-33c00aa3-5870-44c4-b608-3260fec4aef1.png)
